kernel/src/main/java/org/kframework/kompile/Kompile.java

```diff
@@ -447,6 +447,15 @@ public void proverChecksX(Module specModule, Module mainDefModule) {
                 errors.add(KEMException.compilerError("Found syntax declaration in proof module. Only tokens for existing sorts are allowed.", s));
 
         ModuleTransformer mt = ModuleTransformer.fromSentenceTransformer((m, s) -> {
+            if (s instanceof Rule && (s.att().contains(Att.SIMPLIFICATION()))) {
+                KLabel kl = m.matchKLabel((Rule) s);
+                scala.collection.Set<Production> prods = m.productionsFor().get(kl).getOrElse(() -> {
+                    throw KEMException.criticalError("Could not find productions for label " + kl.name(), s);
+                });
```

Comment:
This is probably a somewhat common code pattern? Could we write a function called `getUniqueProductionFor(kl)`, which encapsulates all this logic? Basically it calls `productionsFor().get(kl).getOrElse(() -> { error })` and then asserts that the length of the returned productions in exactly 1, and then returns that unique production? I'm guessing that there are many places in the code using this pattern.

Reply:
Found a way that's even more flexible.